jspwiki-main/src/main/java/org/apache/wiki/tags/AttachmentsIteratorTag.java

```diff
@@ -108,8 +108,9 @@ public final int doStartTag()  {
     public final int doAfterBody() {
         if( bodyContent != null ) {
             try {
-                final JspWriter out = getPreviousOut();
-                out.print(bodyContent.getString());
+                try (JspWriter out = getPreviousOut()) {
```

Comment:
as per [JSPWriter#close](https://docs.oracle.com/javaee/7/api/javax/servlet/jsp/JspWriter.html#close--), this method shouldn't be invoked directly. 

Furthermore, [JSPWriter#flush](https://docs.oracle.com/javaee/7/api/javax/servlet/jsp/JspWriter.html#flush--) states `[...] Once a stream has been closed, further write() or flush() invocations will cause an IOException to be thrown.`, so once this stream is closed, after the custom tags is rendered, the rest of the JSP wouldn't be rendered and will yield an exception? Would you mind verifying if this is the case?

Reply:
> as per [JSPWriter#close](https://docs.oracle.com/javaee/7/api/javax/servlet/jsp/JspWriter.html#close--), this method shouldn't be invoked directly.
> 
> Furthermore, [JSPWriter#flush](https://docs.oracle.com/javaee/7/api/javax/servlet/jsp/JspWriter.html#flush--) states `[...] Once a stream has been closed, further write() or flush() invocations will cause an IOException to be thrown.`, so once this stream is closed, after the custom tags is rendered, the rest of the JSP wouldn't be rendered and will yield an exception? Would you mind verifying if this is the case?

Hey @juanpablo-santos I'll take a look.  I'll let you know. In the meantime, what do you think about leaving the PR open?
TY

Reply:
Hi @juanpablo-santos 

You are absolutely right. After reviewing the JSPWriter documentation and the code again, I agree that closing the JSPWriter stream within the doAfterBody() method could potentially cause an IOException in the subsequent operations by the JSP engine.

I had introduced the try-with-resources to ensure proper resource handling. However, in this context, the JSP engine manages the lifecycle of the JSPWriter stream, and it is best not to interfere with this management.

Therefore, I will revert the try-with-resources change for the JspWriter out stream. Thank you for catching this and for your valuable feedback.